blocks/library/table2/style.scss

```diff
@@ -0,0 +1,36 @@
+.editor-visual-editor__block[data-type="core/table2"] {
```

Comment:
I think we should encourage the use of the new generated classNames for styling `wp-block-table2`